atom/browser/api/atom_api_web_contents.cc

```diff
@@ -1300,11 +1300,17 @@ std::string WebContents::GetUserAgent() {
   return web_contents()->GetUserAgentOverride();
 }
 
-bool WebContents::SavePage(const base::FilePath& full_file_path,
-                           const content::SavePageType& save_type,
-                           const SavePageHandler::SavePageCallback& callback) {
-  auto* handler = new SavePageHandler(web_contents(), callback);
-  return handler->Handle(full_file_path, save_type);
+v8::Local<v8::Promise> WebContents::SavePage(
+    const base::FilePath& full_file_path,
+    const content::SavePageType& save_type) {
+  scoped_refptr<util::Promise> promise = new util::Promise(isolate());
+  auto* handler = new SavePageHandler(web_contents(), promise);
+  bool success = handler->Handle(full_file_path, save_type);
```

Comment:
_(nit)_ `success` is slightly misleading; it could be read as "saved" -- especially when read right before the the "failed to save" message a couple of lines down.

Clearer:

> const bool saveStarted = handler->Handle(...)
> if (!saveStarted)